add data processing page (#421)
moj-analytical-services/user-guidance: source/documentation/tools/tools-for-data-processing.md

> *   Easier to migrate code to [Airflow](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/airflow/) (see below) or send to others to run if they have access to same tools

> ❌ Cons
> *   Initial setup time of projects is time-consuming
> *   Code to run simple SQL is more complex than required on Athena workbench   

### Amazon Athena Console

Another option is to use the [Athena Console](https://user-guidance.analytical-platform.service.justice.gov.uk/data/curated-databases/amazon-athena/) directly on AWS through the Analytical Platform Control Panel. The Athena Console allows you to write and run SQL code directly. This is very handy when exploring data that is already in an Athena database and table.

> ✅ Pros
> * Quick to get started as you don’t need to install anything
> * Creates modular SQL script thinking needed for [Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) (see below)

> ❌ Cons
> * SQL only - limiting how complex your processing can be
> * No proper version control, makes it hard for others to replicate

> > * Data has to be in an Athena database already 

Reviewer: Not necessarily

Author: Which bit? All of it have caveats do they?

Reviewer: You can create databases and tables using Athena e.g. [create table](https://docs.aws.amazon.com/athena/latest/ug/create-table.html)

Author: Resolved by removing

>  

Reviewer: How about adding a section on create-a-derived-table? You don't have to use for orchestration, it also makes it easier to run one-off queries:

```suggestion
 ### Create-a-derived-table
Create-a-derived-table is an in-house developed service that brings  [dbt](https://docs.getdbt.com/docs/introduction), Git, and data access controls together to allow you to deploy derived and processed tables from data available on the Analytical Platform. This extends SQL with features commonly associated with programming languages, enabling more flexible, scalable, and maintainable code. dbt can also assure the quality of your transformations through [data tests](https://docs.getdbt.com/docs/build/data-tests), for example to check whether columns contain null values.
```

## Processing data using an automated pipeline

### Airflow

Apache [Airflow](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/airflow/) is an open-source platform used to programmatically author, schedule, and monitor complex workflows and pipelines. It can be used to run scripts written in R, Python, or any other coding language. Airflow jobs are created using the [airflow repository](https://github.com/moj-analytical-services/airflow). See the [user guidance page on Airflow](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/airflow/) for more information.

> ✅ Pros
> * Well-established and good community support from AP users and wider (used in many organisations)
> * Highly flexible, such as the ability to use GPUs if needed
> * Data can be both structured, and unstructured

> ❌ Cons
> * Have to set up Dockerfile and DAG
> * Have to describe task order in DAG manually
> * Difficult to trigger run only when underlying data has been updated - usually have to hard-code a start time
> * Debugging and logging aren’t great
> * Setup even for simple tasks can be complex

### Create a Derived Table

> [Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) is an in-house developed service that brings dbt, Git, and data access controls together to allow you to deploy derived and processed tables from data available on the Analytical Platform; straight to Athena, in a reproducible way, and with scheduled table updates.

Reviewer: ```suggestion
[Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) also allows you to run scheduled table updates.
```
